### PR TITLE
Feature/dto with main

### PR DESCRIPTION
**Pull request by Hamza-Iqb**
DTO class has been added.

Each rest endpoint in the controller class uses the modelmapper to ensure the parameters given is of the dto object. Which will then be mapped to the entity for the service class to handle.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -56,13 +57,18 @@
...
 		</dependency>
-        <dependency>
-            <groupId>com.vaadin.external.google</groupId>
-            <artifactId>android-json</artifactId>
-            <version>0.0.20131108.vaadin1</version>
-            <scope>compile</scope>
-        </dependency>
-    </dependencies>
+		<dependency>
+			<groupId>com.vaadin.external.google</groupId>
```
Comment (darthrevanyunka) [inline pom.xml:61]: Is it not possible to use GSON instead?

**Review by darthrevanyunka — COMMENTED** [review]
Please review comment

File: pom.xml (hunk 1 above)
Comment (Hamza-Iqb) [reply pom.xml:61]: @darthrevanyunka 
I think this came with the project beforehand. I can change it now, if you would prefer gson

**Review by darthrevanyunka — APPROVED** [review]

**Comment by darthrevanyunka** [comment]
@Hamza-Iqb please resolve merge conflicts. Perhaps chat to Hanno to ensure you aren't overwriting each other's stuff

**Comment by Hamza-Iqb** [comment]
Note: Conflicts have been dealt with.

**Pull request merged**